File: Solutions/2053- Kth Distinct String in an Array/Solution.py
class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        my_dict = Counter(arr)

        for item in arr:
            if my_dict[item] == 1:
                if k == 1:
                    return item
                else:
                    k -= 1
        return ""
        

File: Solutions/2053- Kth Distinct String in an Array/Solution.java
class Solution {
    public String kthDistinct(String[] arr, int k) {
        Map<String, Integer> map = new HashMap<>();
        for (String item : arr) {
            if (!map.containsKey(item)) {
                map.put(item, 1);
            } else {
                map.put(item, map.get(item) + 1);
            }
        }

        for (String item : arr) {
            if (map.get(item) == 1) {
                if (k == 1) {
                    return item;
                } else {
                    k--;
                }
            }
        }

        return "";
    }
}